cloudinit/distros/netbsd.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (C) 2019-2020 Gonéri Le Bouder
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import crypt
+import os
+import platform
+import six
+
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import ssh_util
+from cloudinit import util
+from cloudinit.distros import bsd_util
+
+LOG = logging.getLogger(__name__)
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/rc.conf'
+    ci_sudoers_fn = '/usr/pkg/etc/sudoers.d/90-cloud-init-users'
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'netbsd'
+        cfg['ssh_svcname'] = 'sshd'
+
+    def _read_system_hostname(self):
+        sys_hostname = self._read_hostname(filename='/etc/rc.conf')
+        return ('/etc/rc.conf', sys_hostname)
+
+    def _read_hostname(self, filename, default=None):
+        return bsd_util.get_rc_config_value('hostname')
+
+    def _write_hostname(self, hostname, filename):
+        bsd_util.set_rc_config_value('hostname', hostname,
+                                     fn='/etc/rc.conf')
+
+    def create_group(self, name, members):
+        group_add_cmd = ['groupadd', name]
+        if util.is_group(name):
+            LOG.warning("Skipping creation of existing group '%s'", name)
+        else:
+            try:
+                util.subp(group_add_cmd)
+                LOG.info("Created new group %s", name)
+            except Exception as e:
+                util.logexc(LOG, "Failed to create group %s", name)
+                raise e
```

Comment:
since we're not logging any part of the exception, we don't need to bind it to `e`.

Reply:
from https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

> If no expressions are present, `raise` re-raises the last exception that was active in the current scope. If no exception is active in the current scope, a [`RuntimeError`](https://docs.python.org/3/library/exceptions.html#RuntimeError) exception is raised indicating that this is an error.

Reply:
Indeed, good point. The problem comes initially form the freebsd distro file. I just pushed a PR to clean that up: https://github.com/canonical/cloud-init/pull/166